File: backend/src/main/java/dev/gabus/mi/agua/service/ReciboService.java
package dev.gabus.mi.agua.service;

import dev.gabus.mi.agua.dto.ReciboDTO;
import dev.gabus.mi.agua.model.enums.PaymentStatus;
import java.util.List;

public interface ReciboService {
    ReciboDTO obtenerPorId(Long id);
    List<ReciboDTO> listarPorPropiedad(Long propiedadId);
    List<ReciboDTO> listarPorPropietario(Long usuarioId);
    List<ReciboDTO> listarPorEstado(PaymentStatus estado);
}

File: frontend/src/app/services/recibo.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import { Recibo, PaymentStatus } from '../models/interfaces';

@Injectable({
  providedIn: 'root'
})
export class ReciboService {
  private apiUrl = `${environment.apiUrl}/recibos`;

  constructor(private http: HttpClient) { }

  obtenerPorId(id: number): Observable<Recibo> {
    return this.http.get<Recibo>(`${this.apiUrl}/${id}`);
  }

  listarPorPropiedad(propiedadId: number): Observable<Recibo[]> {
    return this.http.get<Recibo[]>(`${this.apiUrl}/propiedad/${propiedadId}`);
  }

  listarPorPropietario(usuarioId: number): Observable<Recibo[]> {
    return this.http.get<Recibo[]>(`${this.apiUrl}/propietario/${usuarioId}`);
  }

  listarPorEstado(estado: PaymentStatus): Observable<Recibo[]> {
    return this.http.get<Recibo[]>(`${this.apiUrl}/estado/${estado}`);
  }
}
